client/state/themes/selectors/does-theme-require-atomic-site.ts

```diff
@@ -0,0 +1,14 @@
+import { isExternallyManagedTheme } from './is-externally-managed-theme';
+import 'calypso/state/themes/init';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Checks if a theme requires an Atomic site.
+ *
+ * @param {AppState} state   The app global state
+ * @param {string}   themeId The theme ID
+ * @returns {boolean} True if the theme requires an Atomic site
+ */
+export function doesThemeRequireAtomicSite( state: AppState, themeId: string ): boolean {
```

Comment:
I think this function isn't needed. I think it's more clear to remove it and just call `isExternallyManagedTheme` in the one place we're calling `doesThemeRequireAtomicSite`.

My concern here is that there are other themes that also require Atomic (like the WooCommerce themes) but we're not checking for that case in this function.

Since the WooCommerce themes have their own process for transferring the site to Atomic, we don't need the check here and can get rid of this new function.

Reply:
Hey @markbiek, the idea of this function was to support multiple types of themes in the future, instead of checking specifically whether it is managed externally. With that in mind, let me know what you think about it.

Reply:
I think the time to add this function would be when we have more than one type of theme we're checking for or if we need to perform that check in more than one place.

Right now, I think it just adds the potential for confusing and it'd be better to remove it.